OpenRA.Mods.Common/Traits/World/Pathfinder/CellInfoLayerManager.cs

```diff
@@ -0,0 +1,112 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public interface ICellInfoLayerManager
+	{
+		/// <summary>
+		/// Gets a CellLayer of Nodes from the pool
+		/// </summary>
+		CellLayer<CellInfo> GetFromPool();
+
+		/// <summary>
+		/// Puts a CellLayer into the pool
+		/// </summary>
+		void PutBackIntoPool(CellLayer<CellInfo> ci);
+
+		/// <summary>
+		/// Creates (or obtains from the pool) a CellLayer given a map
+		/// </summary>
+		/// <param name="map"></param>
+		/// <returns></returns>
```

Comment:
You can omit any XML elements that are empty, since they're just clutter if there's no worthwhile comment to put there.